document/document.go

```diff
@@ -86,16 +86,107 @@ func (fb FieldBuffer) GetByField(field string) (Value, error) {
 	return Value{}, ErrFieldNotFound
 }
 
+// setFieldValue replaces a field if it already exists or creates one if not.
+func (fb *FieldBuffer) setFieldValue(field string, reqValue Value) error {
+	_, err := fb.GetByField(field)
+	switch err {
+	case ErrFieldNotFound:
+		fb.Add(field, reqValue)
+		return nil
+	case nil:
+		_ = fb.Replace(field, reqValue)
+		return nil
+	}
+
+	return err
+}
+
+// setValueAtPath deep replaces or creates a field
+// through the value path to get the value and create or replace it.
+func (fb *FieldBuffer) setValueAtPath(v Value, p ValuePath, newValue Value) (Value, error) {
+
+	switch v.Type {
+	case DocumentValue:
+		var buf FieldBuffer
+		err := buf.Copy(v.V.(Document))
```

Comment:
On line 174, we are already copying the entire Document, I don't think we need to do it here right?
I think replacing l174 and this line by `buf.ScanDocument` instead will be more efficient as it won't decode values that are untouched

Reply:
I didn't know if the `ScanDocument` function was enough.